typings/styled-components.d.ts

```diff
@@ -18,8 +18,11 @@ export type OuterStyledProps<P> = ThemedOuterStyledProps<P, any>;
 
 export type Interpolation<P> = FlattenInterpolation<P> | ReadonlyArray<FlattenInterpolation<P> | ReadonlyArray<FlattenInterpolation<P>>>;
 export type FlattenInterpolation<P> = InterpolationValue | InterpolationFunction<P>;
-export type InterpolationValue = string | number | StyledComponentClass<any, any>;
+export type InterpolationValue = string | number | Styles | StyledComponentClass<any, any>;
 export type SimpleInterpolation = InterpolationValue | ReadonlyArray<InterpolationValue | ReadonlyArray<InterpolationValue>>;
+export interface Styles {
+  [ruleOrSelector: string]: string | Styles;
```

Comment:
Just curious, does the style only allowed to be a string. Are any other types accepted, like numbers, undefined, nulls, and so on?

Reply:
It's not allowed to be a number, because sc does not allow unit less styles, but it might allow null or undefined. I'll have to check when I'm at my computer. Nice catch 👍

Reply:
I have investigated this some more, and while it "works" to give it anything, SC will just stringify anything that isn't an object and the result will most likely be an invalid property value. If we allow numbers I think that would make people believe that the numbers will be automatically converted, like they would within the style prop of react elements.

Reply:
@nbostrom Let's get this change in; However, we'll need `number` to be added here. `InterpolationValue` contains numbers, so the same must be accepted for object values. :wink:

cc @Igorbek

Reply:
I'm ok with either way: with or without numbers. If numbers are really disallowed (or, going to be deprecated), than let's not include them.

Reply:
@Igorbek they're neither disallowed nor deprecated. You can use them in your non-object interpolations as well, which is why it makes sense to add them to the types here as well

Reply:
sure, let's add them. @nbostrom could you please add `number` here.